### PR TITLE
Create Priority_Queue.dart

### PR DESCRIPTION
**Pull request by Triaro**
Priority Queue (Data Structure)

# Welcome to Dart community

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by Triaro** [comment]
Please add Hactoberfest-accepted label. 
Thank You.

File: data_structures/Queue/Priority_Queue.dart [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+class QueueItem<T> {
+  T item;
+  int priority;
+
+  QueueItem(this.item, this.priority);
+
+  String toString() {
+    return '$item - $priority';
+  }
+}
```
Comment (Parowicz) [inline data_structures/Queue/Priority_Queue.dart:62]: Can i add simple example?
```suggestion
}

void main()
{
  PriorityQueue<int> queue = new PriorityQueue();
  queue.enqueue(1, 2);
  queue.enqueue(2, 1);
  queue.enqueue(3, 3);
  queue.enqueue(4, 2);
  
  
  print(queue.dequeue());
  print(queue.dequeue());
  print(queue.dequeue());
  print(queue.dequeue());
}
```

Comment (Triaro) [reply data_structures/Queue/Priority_Queue.dart:62]: Yes you can. Please!

**Comment by Parowicz** [comment]
Looks Great!

**Comment by Triaro** [comment]
Thank you @Parowicz

**Pull request merged**